app/components/Button.tsx

```diff
@@ -1,38 +1,44 @@
+import type { ActionFunction } from '@remix-run/node';
 import { Icon } from '~/types/Icon';
 
 interface ButtonProps {
+  type: 'submit' | 'reset';
   size: 'small' | 'medium' | 'large';
   label: string;
   varient: 'primary' | 'secondary';
   disabled?: boolean;
   handleClick?: () => void;
 }
 
+export const action: ActionFunction = async ({ request, params }) => {
+  console.log('request:', request);
+};
+
 export const Button: React.FC<ButtonProps> = ({
   size = 'medium',
   label,
   varient = 'primary',
   disabled = false,
-  handleClick,
+  type = 'submit',
```

Comment:
we will need the `handleClick` as this is a generic button which can be used for usecases other than api call.

Reply:
added it back